### DISCUSSION TITLE
The new Task List (Beta)

### DISCUSSION THREAD

**Discussion by hashedin-pranav** [community/community#4251] [Projects and Issues] [answered]
Can we include the title of issue tagged with task in the project board too? 
I can see that in issue description when we open the issue but on other hand it's missing in the issue summary right drawer in the project board.

**Comment by labrose** [comment, marked as answer]
Thanks for the question @hashedin-pranav !

You currently can't view the tracking issue information on the project board, but it sounds like you would request something similar to this idea:

- https://github.com/github/feedback/discussions/4257

You should go ahead and upvote the discussions above! Meanwhile I'm going to mark this one as answered so we can consolidate the discussion in one place.

**Reply by hashedin-pranav** [reply]
Thanks for the update @labrose, I have upvoted the above discussion.

**Reply by Andre601** [reply]
Is that this new task list thing?
I don't quite understand it yet, but I assume you write `- [ ] #:id` and it connects to the issue, PR or discussion?

**Comment by Evavic44** [comment]
It's cool, I love the new progress circle. One thing I would like to see though is an `All tasks completed` message when you check every box.
![IMG_20210623_203348](https://user-images.githubusercontent.com/62628408/123157775-ddeb4900-d462-11eb-851d-f0bc138f1a0f.jpg)

**Comment by princestha** [comment]
Fantastic update. Thanks!!

**Comment by siriwatknp** [comment]
Would be nice to have tasks completed count like `5/8 tasks done`. (I would like to see how much left)

if they are all done, it will be `8 tasks all done` (I still would like to see how many of the tasks)

**Reply by Janmm14** [reply]
`all 8 tasks done`

**Reply by labrose** [reply]
When you use task lists there should be a progress summary in the header of the issue, as well as on the project board and issue index page:

https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists#about-issue-task-lists

Is there somewhere else you were looking for that information or maybe for some reason it's not showing up for you? Could be a bug 😄

**Comment by bryanbarnard** [comment]
Would be nice to have the ability to number the items in the list.

**Reply by vijayiyer05** [reply]
Agreed! It's natural to reference the specific items in comments thread. I guess dynamic links would be needed since the tasks are meant to be easily re-ordered (when they appear in a group).

**Comment by nsamanta** [comment]
I really like this update! Excellent feature! The progress circle along with the count of tasks completed vs total is helpful. 

Here are some suggestions-
1. I would also like to see a `green check` or `green box` saying `180 of 180 tasks completed` or something like that. 
2. Maybe just change the progress circle to green? It would be interesting to see how you manage that if the task list is modified to add or delete some tasks etc 
3. In general, I like the flexibility of moving the task list around because I often like to categorize them, add new ones and move them up or down when they are complete vs incomplete vs ongoing. But many times I lose counts because we have to manually add counts.

**Comment by tjallingvanderwal** [comment]
We usually already have the titles of issues/pull-request in task lists. Now all existing task list are full of duplicate content.

The titles we use in a task list are usually "project internal PoV", whilst PR titles are "repository global PoV". Still a lot of duplication.

**Reply by tjallingvanderwal** [reply]
Additionally, the automatic titles are not visible (obviously) while editing. This makes it harder to reorganise longer task lists.

**Comment by nsamanta** [comment]
I really like that now when you tag another issue (#) you see the issue title. But the Issue titles and comments have the same tag. Is there a way to differentiate that?

**Comment by martin-braun** [comment]
It's a nice feature, but it's not applicable in any situation. For instance, sometimes templates include checkboxes to select the kind of issue. For instance, check this issue I made: https://github.com/FreeTubeApp/FreeTube/issues/1457

Here the issue template wanted me to check the appropriate checkbox, yet it thinks we are dealing with "tasks", when we are not, in fact.

Is it possible to make checkboxes without triggering the "task" feature?

**Reply by denelon** [reply]
Similar asks at https://github.com/github/feedback/discussions/4319 and and https://github.com/github/feedback/discussions/4318.

**Reply by ElijahLynn** [reply]
Tasks really should be first class citizens and be structured data outside of the markdown field, I think. And we should be able to add new tasks, delete tasks, or modify existing tasks without dropping into the markdown editor as there is too much friction and the number one thing we need is a fast system to help make tasking (breaking down work) easy.

**Comment by vielmetti** [comment]
I really like that the new task list shows the status of tasks that are referenced, so that I don't have to 
click through to see if something has been completed.

**Comment by odrori** [comment]
The completed task indicator is nice.
The issue number is usually presented in a sentence as a reference and the appearance of the full name just interrupts the reading flow - Too much info.

**Comment by ratijas** [comment]
"Convert to issue" pop-up button (how do you technically call that invisible-until-mouse-hover thing?) is horrifying. Too easy to accidentally mis-click and create low-quality issues. It doesn't even ask for confirmation — wtﬀ with that?

**Comment by Kilars** [comment]
I would like the ability to mark a task as checked from a commit message, perhaps using issue id and task index. Then the tasks can also work as "sub-issues".
